lxd/db/node.go

```diff
@@ -43,32 +43,50 @@ const ClusterRoleEventHub = ClusterRole("event-hub")
 // ClusterRoleOVNChassis represents a cluster member who operates as an OVN chassis.
 const ClusterRoleOVNChassis = ClusterRole("ovn-chassis")
 
-// ClusterRoles maps role ids into human-readable names.
+// AutomaticClusterRoles contains roles that are automatically assigned by LXD.
+// These roles are managed internally and cannot be manually added or removed by users.
+var AutomaticClusterRoles = map[int]ClusterRole{
```

Comment:
Suggest storing these in a map keyed on type/class/family of role type (manual and automatic) rather than 2 different variables as it allows iterating the map for all role type/class/families rather than having to have duplicate loops when searching for a particular role name.

E.g.

```go
type ClusterRoleClass int
var ClusterRoles map[ClusterRoleClass]map[int]ClusterRole
```